plugins/module_utils/vmware_rest_client.py

```diff
@@ -125,6 +133,14 @@ def connect_to_vsphere_client(self):
         session = requests.Session()
         session.verify = self.params.get('validate_certs')
 
+        if self.params.get('protocol') and self.params.get('proxy_host') and self.params.get('proxy_port'):
```

Comment:
```suggestion
        protocol = self.params.get('protocol')
        proxy_host = self.params.get('proxy_host')
        proxy_port = self.params.get('proxy_port')
        if not all([protocol, proxy_host, proxy_port]):
```

Reply:
@Akasurde are you suggesting to change the if condition ?

Reply:
oh yes. Sorry for the typo.